### ISSUE TITLE
Fix up Sonobuoy flags gen logic

### ISSUE THREAD

**Issue by liztio** [vmware-tanzu/sonobuoy#481]
The conditional logic for dealing with detecting RBAC and server version in [gen.go](https://github.com/heptio/sonobuoy/blob/master/cmd/sonobuoy/app/gen.go#L94) is very convoluted and non-idiomatic. Consider refactoring.

**Comment by johnSchnake** [comment]
Breadcrumb: Just thinking that this is a larger issue than just the gen.go that maybe warrents some cleanup. Clarifying the boundaries/purposes of some of the packages and simplifying the flow of information (e.g. we have lots of similar types like all the gen/config objects and plugin.Definition/manifest.manifest that make it hard to follow when newer to the codebase)

**Comment by johnSchnake** [comment]
Cleanup is always good but closing this for now; doing some grooming on the issue list in general and closing things if either:
 - they weren't been reported by the community
 - don't have a genuine need to get done

The reason for this is that we tended to use the issue list as a backlog of very minor or wish-list type items which we wish to clear out for now.
